priv/perf/apps/load_test/lib/service/datadog/api.ex

```diff
@@ -80,10 +80,16 @@ defmodule LoadTest.Service.Datadog.API do
         {:ok, events}
 
       {:ok, %{body: body}} ->
-        {:error, body}
+        case retries do
+          0 -> {:error, body}
+          _ -> fetch_events(start_time, end_time, environment, retries - 1)
+        end
 
       {:error, error} ->
-        {:error, error}
+        case retries do
+          0 -> {:error, error}
+          _ -> fetch_events(start_time, end_time, environment, retries - 1)
+        end
```

Comment:
why not provide a header function?
```
def fetch_events(start_time, end_time, environment, 0) do
##return error
```

just extract the query and you won't need to have a case inside a case